CIScripts/Report/GenerateTestReport.Tests.ps1

```diff
@@ -7,69 +7,107 @@ Describe "Generating test report" {
+            Join-Path $OutputDir "pretty_test_report/Index.html" | Should Exist
+            Join-Path $OutputDir "reports-locations.json" | Should Exist
+        }
 
-        $FileContents = Get-Content -Raw (Join-Path $OutputDir "raw_NUnit/foo.xml")
-        NormalizeXmlString $FileContents | Should BeExactly $ExpectedXml
+        It "flattens the xml files" {
+            $ExpectedXml = NormalizeXmlString '
+                <test-results failures="0" inconclusive="0" skipped="0" date="2018-01-01" time="15:00:00">
+                    <test-suite name="inner_suite" type="TestFixture" result="Success">
+                        <results>
+                            <test-case name="test" result="Success" />
+                        </results>
+                    </test-suite>
+                </test-results>
+            '
+ 
+            $FileContents = Get-Content -Raw (Join-Path $OutputDir "raw_NUnit/foo.xml")
+            NormalizeXmlString $FileContents | Should BeExactly $ExpectedXml
+        }
     }
 
-    Context "json" {
-        BeforeAll {
-            [
-                Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseDeclaredVarsMoreThanAssignments",
-                "Json",
-                Justification="PSAnalyzer doesn't understand relations of Pester's blocks.")
-            ]
-            $Json = Get-Content -Raw -Path (Join-Path $OutputDir "reports-locations.json") | ConvertFrom-Json
+
+    Context "multiple xml files" {
+        BeforeEach {
+            New-DummyFile -Path (Join-Path $InputDir "foo.xml")
+            New-DummyFile -Path (Join-Path $InputDir "bar.xml")
+            New-DummyFile -Path (Join-Path $InputDir "baz.xml")
+            Convert-TestReportsToHtml -XmlReportsDir $InputDir -OutputDir $OutputDir
+        }
+
+        It "creates appropriate files" {
+            Join-Path $OutputDir "raw_NUnit/foo.xml" | Should Exist
+            Join-Path $OutputDir "raw_NUnit/bar.xml" | Should Exist
+            Join-Path $OutputDir "raw_NUnit/baz.xml" | Should Exist
+            Join-Path $OutputDir "pretty_test_report/Index.html" | Should Exist
+            Join-Path $OutputDir "pretty_test_report/foo.html" | Should Exist
+            Join-Path $OutputDir "pretty_test_report/bar.html" | Should Exist
+            Join-Path $OutputDir "pretty_test_report/baz.html" | Should Exist
+            Join-Path $OutputDir "reports-locations.json" | Should Exist
         }
+    }
+
 
+    Context "providing a json file to be used by monitoring" {
+        BeforeEach {
+            New-DummyFile -Path (Join-Path $InputDir "foo.xml")
+            Convert-TestReportsToHtml -XmlReportsDir $InputDir -OutputDir $OutputDir
+        }
+        
         It "contains valid path to xml report" {
+            $Json = Get-Content -Raw -Path (Join-Path $OutputDir "reports-locations.json") | ConvertFrom-Json
             $Json.'xml_reports'[0] | Should BeExactly './raw_NUnit/foo.xml'
```

Comment:
Before, the json was tested on multiple xmls, now it's tested on a single-file case. Please change the test so it's tested on multiple xmls, as it was before.

Preferably, the json tests should be run on both single- and multiple-file cases.

Reply:
My bad - I wrote *before* because I thought that https://github.com/Juniper/contrail-windows-ci/pull/30 is already merged. Please sync up with @sodar regarding multiple xml tests.